ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -762,6 +769,14 @@ runDataDiffusion tracers
                 dtTracePublicRootPeersTracer
                 timeout
                 DNS.defaultResolvConf
+                DNSActions {
+                  dnsResolverResource = resolverResource,
+                  dnsAsyncResolverResource = asyncResolverResource,
+#if defined(mingw32_HOST_OS)
+                  dnsNewResolverResource = newResolverResource,
+#endif
+                  dnsLookupAWithTTL = lookupAWithTTL
+                }
```

Comment:
I would be cleaner to export `ioDNSActions :: DNSActions` rather than each its component separately.

Reply:
Good idea!